packages/rn-tester-e2e/test/screenObjects/buttonComponent.screen.js

```diff
@@ -0,0 +1,67 @@
+import Utils from '../helpers/utils';
+
+class ButtonComponentScreen {
+
+    buttonScreenElement = Utils.platformSelect({
+        ios: '[label="Button"]',
+        android: '//android.view.ViewGroup/android.widget.TextView[@text="Button"]',
```

Comment:
we should be able to rely on the props of the components [like here](https://github.com/kelset/react-native-e2e-jest-appium-webdriverio/blob/main/__tests__/basicE2E.test.js#L33):

`const UsernameTextInput = await client.$('~username-textinput');`

this will require a bit of work on the RNTester (adding [this util](https://github.com/kelset/react-native-e2e-jest-appium-webdriverio/blob/main/App.js#L21) and then adding the prop to all components)  itself but it will make the whole thing more scalable.

Reply:
this is the folder we need the files are: https://github.com/facebook/react-native/tree/main/packages/rn-tester/js

Reply:
why not use testID and accessibilityLabel ? so xpath will look like this $(//*[@content-desc="Button1" or @name="Button1"]) it will work for iOS and android. But I would use accessibilityLabels/testID $('~Button1') instead of xpath's. Xpaths lookups is really slow :)

Reply:
@lukutism that's what I'm proposing, you can also read more here https://github.com/kelset/react-native-e2e-jest-appium-webdriverio#notes-on-e2e-how-does-it-work